node/setup.go

```diff
@@ -189,8 +189,16 @@ func logNodeStartupInfo(state sm.State, pubKey crypto.PubKey, logger, consensusL
 	}
 }
 
-func onlyValidatorIsUs(state sm.State, pubKey crypto.PubKey) bool {
+func hasMajorityVotingPower(state sm.State, pubKey crypto.PubKey) bool {
 	if state.Validators.Size() > 1 {
+		for _, val := range state.Validators.Validators {
+			if val.PubKey.Equals(pubKey) {
+				if val.VotingPower >= state.Validators.TotalVotingPower()*2/3 {
+					return true
+				}
+			}
+		}
+
 		return false
```

Comment:
Theoretically this function could be achieved with just this part

Reply:
I don’t quite understand

Reply:
I think in the case that there is only one validator in the set we’d want the outer case too.